### PR TITLE
chore(samples): v2 compatible exit handler sample

### PR DESCRIPTION
**Pull request by Bobgy**
**Description of your changes:**
Changes:
* v2 compatible exit handler sample
* use MLMD to verify the exit handler actually ran
* remove TODO item for legacy exit handler for v2 compatible mode, because it cannot really work regarding parameter/artifact

**Checklist:**
- [ ] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: samples/core/exit_handler/exit_handler_test.py [hunk 1]
```diff
@@ -12,24 +12,52 @@
...
-    #   File "/Users/gongyuan/kfp/pipelines/sdk/python/kfp/compiler/v2_compat.py", line 108, in update_op
-    #     artifact_info = {"fileInputPath": op.input_artifact_paths[artifact_name]}
-    # KeyError: 'GCS path'
-    #
-    # TestCase(
-    #     pipeline_func=pipeline_exit_handler,
-    #     mode=kfp.dsl.PipelineExecutionMode.V2_COMPATIBLE,
-    # ),
-])
+from ...test.util import run_pipeline_func, TestCase, KfpMlmdClient
```
Comment (numerology) [inline samples/core/exit_handler/exit_handler_test.py:24]: nit: maybe import module instead of Class/func?

Comment (Bobgy) [reply samples/core/exit_handler/exit_handler_test.py:24]: good suggestion! I'll handle this later in separate PR

File: samples/core/exit_handler/exit_handler_test.py [hunk 2]
```diff
@@ -12,24 +12,52 @@
...
+
+def verify(
+    argo_workflow_name: str, mlmd_connection_config, run: kfp_server_api.ApiRun,
+    **kwargs
+):
+    t = unittest.TestCase()
+    t.maxDiff = None  # we always want to see full diff
+
+    # t.assertEqual(run.status, 'Succeeded')
```
[the remark this answers is not in the document]
Comment (Bobgy) [reply samples/core/exit_handler/exit_handler_test.py]: Actually, I forgot to uncomment it, thanks for the catch!

**Comment by Bobgy** [comment]
Merging the sample, because it's out for review for a while.

**Pull request merged**